src/GregClient/Utility/FileUtilities.cs

```diff
@@ -79,21 +79,49 @@ public static byte[] GetFileHash(FileInfo fInfo)
             return hashValue;
         }
 
+        /// <summary>
+        /// Gets the full path of the user temporary folder
+        /// </summary>
+        /// <returns>String with the full path of the user temporary folder</returns>
         public static string GetTempFolder()
         {
             return Path.GetDirectoryName(Path.GetTempPath());
         }
 
+        /// <summary>
+        /// Generates a full path under the temporary folder
+        /// </summary>
+        /// <returns>String with the full path</returns>
         public static string GetTempZipPath()
         {
             var tempFolder = GetTempFolder();
-            return Path.Combine(tempFolder, "gregPkg" + DateTime.Now.Millisecond.ToString() + ".zip");
+            string tempZipPath = Path.Combine(tempFolder, String.Format("gregPkg{0}.zip", DateTime.Now.Millisecond.ToString());
+            int range = 1;
+
+            while (File.Exists(tempZipPath))
+            {
+                tempZipPath = Path.Combine(tempFolder, String.Format("gregPkg{0}{1}.zip", DateTime.Now.Millisecond.ToString(), range++));
+            }
+
+            return tempZipPath;
         }
 
+        /// <summary>
+        /// Generates a full output path under the temporary folder
+        /// </summary>
+        /// <returns>String with the full path</returns>
         public static string GetTempZipOutputPath()
```

Comment:
I think since these methods are very similar in implementation we should just have the main logic reside in a `private` helper function:
```
private static string GetTempZipPathHelper(bool output)
{
var tempFolder = GetTempFolder();
            var filename = output ? true : "gregPkgOutput" ? "gregPkg";
            string tempZipPath = Path.Combine(tempFolder, String.Format("{0}{1}.zip", filename, DateTime.Now.Millisecond.ToString());
            int range = 1;

            while (File.Exists(tempZipPath))
            {
                tempZipPath = Path.Combine(tempFolder, String.Format("{0}{1}{2}.zip", filename, DateTime.Now.Millisecond.ToString(), range++));
            }

            return tempZipPath;
}
```

Reply:
Totally agree

Reply:
Resolved on the new commit